core/src/models/Property.ts

```diff
@@ -361,6 +361,25 @@ export class Property extends LoggedModel<Property> {
     );
   }
 
+  @AfterSave
+  static async ensureRecordsMarkedPendingOnTypeChange(instance: Property) {
+    // Check to see if type changed
+    if (instance.changed("type") && instance.state === "ready") {
+      //mark all relevant recordproperties as "pending"
+      await RecordProperty.update(
+        { state: "pending", startedAt: null },
+        { where: { propertyId: instance.id } }
+      );
+
+      const source = await instance.$get("source");
+
+      await GrouparooRecord.update(
+        { state: "pending" },
+        { where: { modelId: source.modelId } }
+      );
```

Comment:
Noting a realization:

We don't need to look up which Records are effected by this property - by definition, it will be all records in the Model